src/main/java/com/google/devtools/build/lib/worker/WorkerSpawnRunner.java

```diff
@@ -286,6 +289,7 @@ private WorkResponse execInWorker(
     try {
       try {
         worker = workers.borrowObject(key);
+        request = request.toBuilder().setRequestId(worker.getWorkerId()).build();
```

Comment:
Can we avoid this copy and variable rewrite somehow? (Can't tell why we have to do this.)

Reply:
Since we need to set request ID, and we need worker ID and request ID to be the same for simplicity. Anyway, I changed it so we construct request after getting a worker.

Reply:
Now it constructs the WorkRequest after borrowing a worker.